src/consumer.rs

```diff
@@ -146,17 +136,105 @@ impl PartitionConsumerJS {
         &self,
         offset: OffsetWrapper,
     ) -> Result<PartitionConsumerIterator, FluvioError> {
-        if let Some(client) = &self.inner {
-            let stream = client.stream(offset.0).await?;
-            let mut iterator = PartitionConsumerIterator::new();
-            iterator.set_inner(Box::pin(stream));
-            Ok(iterator)
-        } else {
-            Err(FluvioError::Other(CLIENT_NOT_FOUND_ERROR_MSG.to_owned()))
+        let client = self
+            .inner
+            .as_ref()
+            .ok_or_else(|| FluvioError::Other(CLIENT_NOT_FOUND_ERROR_MSG.to_string()))?;
+
+        let stream = client.stream(offset.0).await?;
+        let mut iterator = PartitionConsumerIterator::new();
+        iterator.set_inner(Box::pin(stream));
+        Ok(iterator)
+    }
+}
+
+pub struct RecordJS {
+    inner: Option<Record>,
+}
+
+#[node_bindgen]
+impl RecordJS {
+    #[node_bindgen(constructor)]
+    pub fn new() -> Self {
+        Self { inner: None }
+    }
+
+    fn set_inner(&mut self, inner: Record) {
+        self.inner = Some(inner);
+    }
+
+    #[node_bindgen]
+    pub fn key(&self) -> Option<String> {
```

Comment:
This should return ArrayBuffer.   Let's not tied key/value to String

Reply:
Sure. I felt like this could go either way, and it had seemed perhaps reasonable to use strings in the default accessor and to also provide `keyBuffer` accessor for bytes. But I don't have a problem making the default accessor work with bytes. If we're doing this, what if we also add `keyString` and `valueString`?

Reply:
That's possible too. I had originally not gone that route because I didn't want to change node-bindgen if I didn't have to (but now I think I will for producer anyway)